### DISCUSSION TITLE
custom width for switch in chakra UI

### DISCUSSION THREAD

**Discussion by Zack616** [chakra-ui/chakra-ui#7515] [Q&A]
<Switch sx={{ 'span.chakra-switch__track:not([data-checked])': { backgroundColor: 'pink.300' } }} />

in the above example we can set custom bg for switch  . . like the above example can anyone please help me like how can I change the width of the switch in chakra UI? by default we can only use size=sm or md or lg etc . .what I wanted is that I could be able to set custom width like width='30px' of the switch. Is there any way to achieve this ?

**Comment by NairiAreg** [comment]
I guess no way
No answer until now, sad

**Comment by Yewhral** [comment]
You can set the width of a track element and set a matching translation for thumb anatomy part.





    definePartsStyle({
    container: {},    
    track: {
      w: '34px',
      h: '20px',
    },
    thumb: {
      w: '16px',
      h: '16px',
      _checked: {
        transform: 'translateX(14px)',
      },
    },

**Comment by jaworsfold-aviva** [comment]
If anyone else is looking for this. Here's a handy function for defining some of these traits:

```
type TrackProps = {
  text: string;
  fontColor: string;
  backgroundColor: string;
};

export type Track = {
  checked: TrackProps;
  unchecked: TrackProps;
  fontSize?: string;
  fontWeight?: string;
};

function setTrackStyle(isChecked: boolean, track: Track): SystemStyleObject {
  return {
    ".chakra-switch__track::after": {
      content: isChecked
        ? `'${track.checked.text}'`
        : `'${track.unchecked.text}'`,
      color: isChecked ? track.checked.fontColor : track.unchecked.fontColor,
      display: "block",
      position: "absolute",
      transform: "translate(-50%,-50%)",
      lineHeight: "1.5rem",
      top: "1.05rem",
      left: isChecked ? "28%" : "70%",
      fontWeight: track.fontWeight ?? "bold",
      fontSize: track.fontSize ?? "1rem"
    },
    ".chakra-switch__track": {
      background: isChecked
        ? track.checked.backgroundColor
        : track.unchecked.backgroundColor,
      width: "4.375rem",
      height: "2rem"
    },
    ".chakra-switch__thumb": {
      width: "2rem",
      height: "2rem",
      _checked: {
        transform: "translateX(2.36rem)"
      }
    }
  };
```

Then you can just:
```
    <Switch
      isChecked={isChecked}
      onChange={onChange}
      sx={setTrackStyle(isChecked, track)}
      {...rest}
    />
```
